src/codegen.cpp

```diff
@@ -4350,14 +4334,15 @@ static void emit_function(jl_lambda_info_t *lam, jl_llvm_functions_t *declaratio
 
     DebugLoc noDbg;
     ctx.debug_enabled = true;
-    if (dbgFuncName[0] == 0) {
+    if (dbgFuncName.empty()) {
         // special value: if function name is empty, disable debug info
-        builder.SetCurrentDebugLocation(noDbg);
-        ctx.debug_enabled = false;
         do_coverage = false;
         do_malloc_log = false;
+        //dbgFuncName = filename; // for testing, uncomment this line
+        ctx.debug_enabled = dbgFuncName.empty();
```

Comment:
er, yeah that polarity is flipped. this version would mean that debug info is emitted for kwsorters